react/src/views/Transfers/TransferAgreementOverview/TransferAgreementOverview.test.tsx

```diff
@@ -0,0 +1,286 @@
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import { screen, render } from "tests/test-utils";
+import { generateMockTransferAgreement } from "mocks/transferAgreements";
+import { mockGraphQLError, mockNetworkError } from "mocks/functions";
+import { TransferAgreementState, TransferAgreementType } from "types/generated/graphql";
+import TransferAgreementOverviewView, {
+  ACCEPT_TRANSFER_AGREEMENT,
+  ALL_TRANSFER_AGREEMENTS_QUERY,
+  CANCEL_TRANSFER_AGREEMENT,
+  REJECT_TRANSFER_AGREEMENT,
+} from "./TransferAgreementOverviewView";
+
+const mockSuccessfullTransferAgreementsQuery = ({
+  query = ALL_TRANSFER_AGREEMENTS_QUERY,
+  variables = {},
+  state = TransferAgreementState.UnderReview,
+  type = TransferAgreementType.Bidirectional,
+  isInitiator = true,
+}) => ({
+  request: {
+    query,
+    variables,
+  },
+  result: {
+    data: {
+      transferAgreements: [generateMockTransferAgreement({ state, type, isInitiator })],
+    },
+  },
+});
+
+// Toasts are persisting throughout the tests since they are rendered in the wrapper and not in the render.
+// Therefore, we need to close them after each test since we get easily false positives
+// Everywhere where we have more than one occation of a toast we should do this.
+afterEach(() => {
+  const closeToastButtons = screen.queryAllByRole("button", { name: /close/i });
+  closeToastButtons.forEach((button) => userEvent.click(button));
+});
+
+it("4.2.2a - Failed to Fetch Initial Data (GraphQlError)", async () => {
+  render(<TransferAgreementOverviewView />, {
+    routePath: "/bases/:baseId/transfers/agreements",
+    initialUrl: "/bases/1/transfers/agreements",
+    mocks: [mockGraphQLError(ALL_TRANSFER_AGREEMENTS_QUERY)],
+  });
+
+  // Check if Error is shown
+  expect(await screen.findByTestId("ErrorAlert")).toBeInTheDocument();
+  // Check if Table is not shown
+  expect(screen.queryByRole("table")).not.toBeInTheDocument();
+}, 10000);
+
+it("4.2.2b - Failed to Fetch Initial Data (GraphQlError)", async () => {
+  render(<TransferAgreementOverviewView />, {
+    routePath: "/bases/:baseId/transfers/agreements",
+    initialUrl: "/bases/1/transfers/agreements",
+    mocks: [mockNetworkError(ALL_TRANSFER_AGREEMENTS_QUERY)],
+  });
+
+  // Check if Error is shown
+  expect(await screen.findByTestId("ErrorAlert")).toBeInTheDocument();
+  // Check if Table is not shown
+  expect(screen.queryByRole("table")).not.toBeInTheDocument();
+});
+
+it("4.2.1 - Initial Load of Page", async () => {
+  render(<TransferAgreementOverviewView />, {
+    routePath: "/bases/:baseId/transfers/agreements",
+    initialUrl: "/bases/1/transfers/agreements",
+    mocks: [mockSuccessfullTransferAgreementsQuery({})],
+  });
+
+  // 4.2.1.1 - Is the Loading State Shown First?
+  // expect(await screen.findByTestId("TableSkeleton")).toBeInTheDocument();
```

Comment:
Did this not work?

Reply:
it is flaky. react testing library does not recommend the usage of `findByTestId` but I have not figured out a better way.